classport-agent/src/test/java/ClassportAgentTest.java

```diff
@@ -0,0 +1,164 @@
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import org.junit.jupiter.api.Test;
+
+import io.github.chains_project.classport.agent.ClassportAgent;
+import io.github.chains_project.classport.commons.AnnotationReader;
+import io.github.chains_project.classport.commons.ClassportInfo;
+
+public class ClassportAgentTest {
+
+    private static final String RESOURCES_PATH = "src/test/resources/";
+    private static final String CLASSPORT_TREE_PATH = RESOURCES_PATH + "classport-deps/classport-deps-tree";
+    private static final String CLASSPORT_LIST_PATH = RESOURCES_PATH + "classport-deps/classport-deps-list";
+    private static final String ANNOTATED_CLASS_PATH = RESOURCES_PATH + "annotated-classes/StringUtils.class";
+    private static final String NOT_ANNOTATED_CLASS_PATH = RESOURCES_PATH + "not-annotated-classes/StringUtils.class";
+
+    @Test
+    void shouldGenerateDependencyListAndTreeFiles() throws Exception {
+        HashMap<String, ClassportInfo> sbom = getSBOM();
+        
+        // Invokation of private method WriteSBOM of ClassportAgent class
+        invokeWriteSBOMMethod(sbom);
+
+        File treeFile = new File(System.getProperty("user.dir"), "classport-deps-tree");
+        File listFile = new File(System.getProperty("user.dir"), "classport-deps-list");
+        
+        assertAll(
+            () -> assertTrue(treeFile.exists(), "Tree output file should be created"),
+            () -> assertTrue(listFile.exists(), "List output file should be created"),
+            () -> assertTrue(compareFiles(treeFile, new File(CLASSPORT_TREE_PATH)), "Tree files should be identical."),
+            () -> assertTrue(compareFiles(listFile, new File(CLASSPORT_LIST_PATH)), "List files should be identical.")
+        );
+
+        treeFile.delete();
+        listFile.delete();
```

Comment:
Why are you deleting them? They seem to be pushed to our repository as well. I think what you want to do is to generate these tree and list files again and then compare it to the one under `RESOURCE_PATH`. Instead of taking care of deletion of generated files, you could use [`@TempDir` annotation](https://github.com/chains-project/sbom.exe/blob/53df531cb05113f127df2da0b54d566cf149c672/classfile-fingerprint/src/test/java/io/github/algomaster99/terminator/index/RuntimeIndexerTest.java#L15).

Reply:
Discussed in persom. classport creates these files in the project directory. @serenacofano please create an issue to pass a `path` argument.